pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -237,23 +226,23 @@ func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.Cro
 		if r.podSpecChanged(ra.Spec.Template.Spec, expected.Spec.Template.Spec) {
 			ra.Spec.Template.Spec = expected.Spec.Template.Spec
 			if ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Update(ra); err != nil {
-				return ra, err
+				return ra, fmt.Errorf("updating receive adapter: %v", err)
 			}
 			logging.FromContext(ctx).Info("Receive Adapter updated.", zap.Any("receiveAdapter", ra))
-		} else {
-			logging.FromContext(ctx).Info("Reusing existing receive adapter", zap.Any("receiveAdapter", ra))
 		}
+		logging.FromContext(ctx).Debug("Reusing existing receive adapter", zap.Any("receiveAdapter", ra))
 		return ra, nil
 	}
 
-	if ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Create(expected); err != nil {
-		return nil, err
+	ra, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Create(expected)
+	if err != nil {
+		return nil, fmt.Errorf("creating receive adapter: %v", err)
 	}
-	logging.FromContext(ctx).Info("Receive Adapter created.", zap.Any("receiveAdapter", expected))
-	return ra, err
+	logging.FromContext(ctx).Info("Receive Adapter created.", zap.Any("receiveAdapter", ra))
+	return ra, nil
 }
 
-func (r *Reconciler) podSpecChanged(oldPodSpec corev1.PodSpec, newPodSpec corev1.PodSpec) bool {
+func (*Reconciler) podSpecChanged(oldPodSpec corev1.PodSpec, newPodSpec corev1.PodSpec) bool {
```

Comment:
Does this need to be a method of Reconciler then? or it can be just a helper function? 
Whatever the convention is, I'm OK with it.

Reply:
Done.